src/components/cc-tile-metrics/cc-tile-metrics.js

```diff
@@ -0,0 +1,369 @@
+import { i18n } from '../../lib/i18n.js';
+import { tileStyles } from '../../styles/info-tiles.js';
+import { skeletonStyles } from '../../styles/skeleton.js';
+import { ccLink, linkStyles } from '../../templates/cc-link/cc-link.js';
+import '../cc-stretch/cc-stretch.js';
+import '../cc-button/cc-button.js';
+import '../cc-error/cc-error.js';
+
+// TODO: How do we handle this with cc-icon?
+const closeSvg = new URL('../../assets/close.svg', import.meta.url).href;
+const infoSvg = new URL('../../assets/info.svg', import.meta.url).href;
+const grafanaSvg = new URL('../../assets/grafana.svg', import.meta.url).href;
+
+Chart.register(BarController, BarElement, Tooltip, CategoryScale, LinearScale, Title);
+
+const TOP_COLOR_CHART = 'rgb(190, 52, 97)';
+const MIDDLE_COLOR_CHART = 'rgb(78, 100, 234)';
+const BOTTOM_COLOR_CHART = 'rgb(100, 146, 234)';
+
+// TODO: If we agree on this don't forget to move this in CSS.
+const TOP_COLOR_LEGEND = 'var(--cc-color-text-danger)';
+const MIDDLE_COLOR_LEGEND = 'var(--color-blue-80)';
+const BOTTOM_COLOR_LEGEND = 'var(--cc-color-text-primary)';
+
+const TOP_THRESHOLD = 80;
+const BOTTOM_THRESHOLD = 20;
+
+const NUMBER_OF_POINTS = 24;
+
+const SKELETON_REQUESTS = Array
+  .from(new Array(NUMBER_OF_POINTS))
+  .map((_, index) => {
+    const startTs = new Date().getTime();
+    return { value: Math.random(), timestamp: startTs + index * 3600 };
+  });
+
+/**
+ * @typedef {import('./cc-tile-metrics.types.js').TileMetricsState} TileMetricsState
+ */
+
+/**
+ * A "tile" component to display CPU and RAM metrics in a bar chart.
+ *
+ * @cssdisplay grid
+ */
+export class CcTileMetrics extends LitElement {
+
+  static get properties () {
+    return {
+      grafanaLink: { type: String },
+      metricsLink: { type: String },
+      metrics: { type: Object },
+      _docs: { type: Boolean, attribute: false },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {string} - Sets the link to the grafana app. */
+    this.grafanaLink = '';
+
+    /** @type {string} - Sets the link to the metrics section. */
+    this.metricsLink = '';
+
+    /** @type {TileMetricsState} - State of the component. */
+    this.metrics = { state: 'loading' };
+
+    /** @type {boolean} */
+    this._docs = false;
+  }
+
+  _createChart (chartElement) {
+    return new Chart(chartElement, {
+      type: 'bar',
+      options: {
+        maintainAspectRatio: false,
+        radius: 0,
+        interaction: {
+          intersect: false,
+        },
+        scales: {
+          x: {
+            display: false,
+            stacked: true,
+          },
+          y: {
+            display: false,
+            beginAtZero: true,
+          },
+        },
+        plugins: {
+          legend: {
+            display: false,
+          },
+          tooltip: {
+            enabled: false,
+          },
+        },
+      },
+    });
+  }
+
+  _getChartData (inputData) {
+
+    const data = inputData ?? SKELETON_REQUESTS;
```

Comment:
Not sure why but I would prefer `SKELETON_REQUESTS` to be passed to `_getChartData` instead of being implicit with null.